code/datums/components/decals/blood.dm

```diff
@@ -0,0 +1,31 @@
+
+/datum/component/decal/blood
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+/datum/component/decal/blood/Initialize(_icon, _icon_state, _dir, _cleanable=CLEAN_STRENGTH_BLOOD, _color, _layer=ABOVE_OBJ_LAYER)
+	if(isitem(parent))
+		return ..()
+	else
+		. = COMPONENT_INCOMPATIBLE
+		CRASH("Warning: Blood decal attempted to be added to non-item of type [parent.type]")

```

Comment:
just do if(!isitem) crash

It makes it easier to read if it ever becomes necessary to add more stuff here